drift/lib/src/runtime/query_builder/expressions/case_when.dart

```diff
@@ -1,30 +1,56 @@
-import 'package:meta/meta.dart';
-
 import '../query_builder.dart';
 
+/// A `CASE WHEN` expression (without base expression) in sqlite.
+class CaseWhenExpression<R extends Object> extends _CaseWhenExpression<R> {
+  /// Creates a `CASE WHEN` expression from the independent components.
+  CaseWhenExpression({
+    required Map<Expression, Expression<R>> when,
```

Comment:
The keys should have a type of `Expression<bool>` here, right?

For `caseMatch`, we can sort of assume that the order doesn't matter, but for a when expression without a base, it really does. So maybe we should change the parameter of `when` to `List<MapEntry<Expression<bool>, Expression<R>>>` directly?

Reply:
> The keys should have a type of Expression<bool> here, right?   

That's true, will be fixed.

> So maybe we should change the parameter of when.  

No, I think it is a bad idea because of the usage. 
Variant with `List`:
```dart
CaseWhenExpression<int>(when: [
  MapEntry(const CustomExpression("'id' IS 1"), const Constant(1)),
  MapEntry(const CustomExpression("'id' IS 2"), const Constant(2)),
]
```
vs variant with `Map`:
```dart
CaseWhenExpression<int>(when: {
  const CustomExpression("'id' IS 1"): const Constant(1),
  const CustomExpression("'id' IS 2"): const Constant(2),
}),
```

The order will be described in the doc

Reply:
My point is that `Map`s aren't guaranteed to keep their order when we call `.entries`. It happens that the default map implementation in Dart (`LinkedHashMap`) does preserve insertion order, but it's not guaranteed that map literals will always use that type. Users might also pass a different map (e.g. a `HashMap` they got from somewhere) which then uses another order and subtly breaks the generated SQL.

In the future, we'll be able to use records for something like:

```dart
CaseWhenExpression<int>(when: [
  (const CustomExpression("'id' IS 1"), const Constant(1)),
  (const CustomExpression("'id' IS 2"), const Constant(2)),
]),
```

Until then, I think we should use a list directly since the order of entries is important here.